src/AppBundle/Twig/SourceCodeExtension.php

```diff
@@ -103,9 +110,15 @@ private function getTemplateCode()
      * Twig_Loader_Filesystem (e.g. TwigBundle:Exception:exception.txt.twig notation
      * or an anonymous template created by the {% embed %} tag).
      */
-    private function getTemplateRelativePath()
+    private function getTemplateFilePath()
     {
-        return 'app/Resources/views/'.$this->template->getTemplateName();
+        $projectBaseDir = realpath($this->kernelRootDir.'/..');
+        $absolutePath = $this->kernelRootDir.'/Resources/views/'.$this->template->getTemplateName();
+        $relativePath = ltrim(str_replace($projectBaseDir, '', $absolutePath), DIRECTORY_SEPARATOR);
+
+        $fileLink = strtr($this->fileLinkFormat, array('%f' => $absolutePath, '%l' => 1));
+
+        return empty($fileLink) ? $relativePath : new \Twig_Markup(sprintf('<a href="%s">%s</a>', $fileLink, $relativePath), 'UTF-8');
```

Comment:
I suggest extracting the logic building a relative path into a separate private method to avoid duplicating it